emissionsapi/preprocess.py

```diff
@@ -25,20 +26,28 @@
 
 @emissionsapi.db.with_session
 def list_ncfiles(session):
-    """Generator yielding all nc files in download path.
+    """list all nc files in storage.
+
+    :param session: SQLAlchemy Session
+    :type session: sqlalchemy.orm.session.Session
+    :return: Set of all unprocessed files
+    :rtype: set
     """
-    # Iterate through the files and directories in storage path
-    for f in os.listdir(storage):
-        # Check if file was already added
-        if session.query(emissionsapi.db.File)\
-                  .filter(emissionsapi.db.File.filename == f).count():
-            logger.info("Skipping %s", f)
-            continue
-        # Join directory and filename
-        filepath = os.path.join(storage, f)
-        # yield file ending with '.nc'
-        if os.path.isfile(filepath) and filepath.endswith('.nc'):
-            yield filepath
+    # Get all *.nc files as set
+    files = {x for x in glob.glob(os.path.join(storage, "*.nc"))}
+    logger.info('%s nc files to process', len(files))
+
+    # Get all already processes files from the database as set
+    processed_files = {
+        os.path.join(storage, x.filename) for x in session.query(
+            emissionsapi.db.File.filename)}
+    logger.info('%s nc files already processed', len(processed_files))
```

Comment:
```suggestion
    logger.info('%s nc files already processed in total', len(processed_files))
```

…makes it a bit more clear that it's not the amount of files part of the previous mentioned files that have been processed.